diff --git a/vendor/github.com/go-git/go-git/v5/worktree_zos.go b/vendor/github.com/go-git/go-git/v5/worktree_zos.go
new file mode 100644
--- /dev/null
+++ b/vendor/github.com/go-git/go-git/v5/worktree_zos.go
@@ -0,0 +1,27 @@
+//go:build zos
+// +build zos
+
+package git
+
+import (
+	"syscall"
+	"time"
+
+	"github.com/go-git/go-git/v5/plumbing/format/index"
+)
+
+func init() {
+	fillSystemInfo = func(e *index.Entry, sys interface{}) {
+		if os, ok := sys.(*syscall.Stat_t); ok {
+			e.CreatedAt = time.Unix(os.Ctim.Unix())
+			e.Dev = uint32(os.Dev)
+			e.Inode = uint32(os.Ino)
+			e.GID = os.Gid
+			e.UID = os.Uid
+		}
+	}
+}
+
+func isSymlinkWindowsNonAdmin(_ error) bool {
+	return false
+}
